mempool/src/interface/mempool_interface.rs

```diff
@@ -25,7 +26,11 @@ use subsystem::{CallRequest, ShutdownRequest};
 
 pub trait MempoolInterface: Send + Sync {
     /// Add a transaction to mempool
-    fn add_transaction(&mut self, tx: SignedTransaction) -> Result<TxStatus, Error>;
+    fn add_transaction(
+        &mut self,
+        tx: SignedTransaction,
+        origin: TxOrigin,
```

Comment:
I would call it "Source", since we also have BlockSource. Just for consistency.

Reply:
There already is `TransactionSource` in the tx verifier, it's used for something else. It's probably better to avoid unifying the two although there are some similarities.

Reply:
I see.